src/kakarot/instructions/system_operations.cairo

```diff
@@ -204,32 +214,17 @@ namespace SystemOperations {
         // 1 - offset: byte offset in the memory in bytes
         let (stack, popped) = Stack.pop_n(self=stack, n=2);
 
-        let size = popped[0];
-        let offset = popped[1];
+        let offset = popped[0];
```

Comment:
This change was the crux of the issues I was seeing with revert reason propagating from sub to calling context.